otx/algorithms/common/adapters/torch/utils/automatic_bs.py

```diff
@@ -0,0 +1,89 @@
+"""Algorithm to find a proper batch size which is fit to current GPU device."""
+
+# Copyright (C) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+from typing import Callable
+
+import torch
+
+from otx.algorithms.common.utils.logger import get_logger
+
+logger = get_logger()
+
+
+def adapt_batch_size(train_func: Callable[[int], None], default_bs: int, trainset_size: int) -> int:
+    """Decrease batch size if default batch size isn't fit to current GPU device.
+
+    Args:
+        train_func (Callable[[int], None]): training function with single arugment to set batch size
+        default_bs (int): default batch size
+        trainset_size (int): training dataset size
+
+    Returns:
+        int: proper batch size possibly decreased as default value isn't fit
+    """
+    if default_bs <= 0:
+        raise ValueError("Batch size should be bigger than 0.")
+    if trainset_size <= 0:
+        raise ValueError("train data set size should be bigger than 0.")
+
+    if trainset_size < default_bs:
+        default_bs = trainset_size
```

Comment:
Why don't you create a new variable to keep `current` or `taget` batch size rather than changing the input arg? Default should be always default value.

Reply:
You're right. I changed it to `current_bs`